cli/server.go

```diff
@@ -813,6 +814,26 @@ func (r *RootCmd) Server(newAPI func(context.Context, *coderd.Options) (*coderd.
 				options.SwaggerEndpoint = cfg.Swagger.Enable.Value()
 			}
 
+			batchStatsTicker := time.NewTicker(29 * time.Second) // Hard-coding.
+			defer batchStatsTicker.Stop()
+			b, err := batchstats.New(
+				batchstats.WithLogger(options.Logger.Named("batchstats")),
+				batchstats.WithStore(options.Database),
+				batchstats.WithTicker(batchStatsTicker.C),
+			)
+			if err != nil {
+				return xerrors.Errorf("failed to create agent stats batcher: %w", err)
+			}
+			options.StatsBatcher = b
+			batcherDone := make(chan struct{})
+			go func() {
+				b.Run(ctx)
+				close(batcherDone)
+			}()
+			defer func() {
+				<-batcherDone
+			}()
```

Comment:
nit: maybe you should consider wrapping into a two-liner: function call with `go func()` inside, and `defer closeFunc`

Reply:
Should be fixed with updated constructor that returns a closer